[NIFIREG-13] Initial implementation of the registry UI/UX

scottyaslan opened the pull request:
You may view the FDS atomic component NgModule demo here: http://localhost:8080/nifi-registry/fluid-design-system

The initial NiFi Registry UI can be located at the following: http://localhost:8080/nifi-registry/explorer/grid-list/23f6cc59-0156-1000-06b4-2b0810089090

mcgilman commented:
Will review...

mcgilman commented on nifi-registry-web-ui/src/main/webapp/components/administration/users/nf-registry-users-administration.js:
```diff
@@ -0,0 +1,128 @@
+                this.router.navigateByUrl('/nifi-registry/administration/' + this.nfRegistryService.registry.id + '/users(sidenav:user/add)');
+                break;
+            default:
+                this.router.navigateByUrl('/nifi-registry/administration/' + this.nfRegistryService.registry.id + '/users(' + action.type + ':user/' + action.name + '/' + user.id + ')');
```
Does `navigateByUrl` automatically encode these URI components?

scottyaslan replied:
Yes

mcgilman commented:
The PR is a great starting point for the registry front end. Obviously, there is some work remaining to get the front end and back end integrated but we can file subsequent JIRAs for that effort and coordinate that work among the relevant contributors. 

Also a couple more general comments:

- We'll need to figure out where the fluid design system will live and then move the demo application with it. I do not believe it belongs as part of the registry front end. But, let's file a follow-on JIRA for this effort.
- There appears to be some delay when loading the page for the first time. Let's file another follow-on JIRA to investigate further and possibly add a loading screen.

Let's address a few of the smaller items mentioned here and get this merged so we can continue with the integration effort while looking into a few of the larger term items which we'll file follow-on JIRAs for.

Thanks again @scottyaslan!

scottyaslan commented on nifi-registry-web-ui/src/main/platform/core/package.json:
```diff
@@ -0,0 +1,33 @@
+  "scripts": {
+  },
+  "engines": {
+    "node": ">=6.11.1",
```
I removed this file for now as it is unnecessary. It would be for defining the FDS NgModule as an npm package.

scottyaslan commented on nifi-registry-web-ui/src/main/frontend/package.json:
```diff
@@ -0,0 +1,65 @@
+  },
+  "dependencies": {
+    "@covalent/core": "^1.0.0-beta.6",
+    "angular2-moment": "^1.6.0",
```
def

scottyaslan commented on nifi-registry-web-ui/src/main/platform/core/LICENSE.md:
```diff
@@ -0,0 +1,22 @@
+The MIT License
```
I removed this file until we can sort it out.

scottyaslan commented on nifi-registry-web-ui/src/main/webapp/nf-registry-bootstrap.js:
```diff
@@ -0,0 +1,57 @@
+    $.ajax({
+        url: translationFile,
+        dataType: "text",
+        success: function (translations) {
```
ok

scottyaslan commented on nifi-registry-web-ui/src/main/webapp/nf-registry.routes.js:
```diff
@@ -0,0 +1,104 @@
+    }]
+    // canActivate: [AuthGuard] //TODO: https://angular.io/api/router/CanActivate https://scotch.io/tutorials/routing-angular-2-single-page-apps-with-the-component-router
+}, {
+    path: 'nifi-registry/fluid-design-system',
```
https://issues.apache.org/jira/projects/NIFIREG/issues/NIFIREG-14?filter=allopenissues

scottyaslan commented:
@mcgilman I have addressed or replied to your comments, rebased against master, and updated this PR. I also opened two JIRA: https://issues.apache.org/jira/browse/NIFIREG-14 to address the FDS NgModule as well as https://issues.apache.org/jira/browse/NIFIREG-15 to address the splash screen and page load times.

mcgilman commented:
Thanks for updating @scottyaslan! This has been merged to master.
